catalyst-toolbox-cli/src/cli/notifications/responses/create_message.rs

```diff
@@ -0,0 +1,60 @@
+use reqwest::StatusCode;
+use serde::de::Error;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+#[derive(Serialize, Deserialize)]
+pub struct InnerResponse {
+    #[serde(alias = "Messages")]
+    messages: Vec<String>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateMessageResponse {
+    #[serde(
+        deserialize_with = "deserialize_status_code",
+        serialize_with = "serialize_status_code"
+    )]
+    status_code: StatusCode,
+    status_message: String,
+    response: InnerResponse,
+}
+
+fn deserialize_status_code<'de, D>(deserializer: D) -> Result<StatusCode, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    StatusCode::from_u16(u16::deserialize(deserializer)?)
+        .map_err(|_| D::Error::custom("Invalid StatusCode"))
+}
+
+fn serialize_status_code<S>(status_code: &StatusCode, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: Serializer,
+{
+    serializer.serialize_u16(status_code.as_u16())
+}
+
+#[cfg(test)]
+mod test {
+    use super::CreateMessageResponse;
+    use reqwest::StatusCode;
+
+    #[test]
+    fn test_deserialize() {
+        let msg = r#"{
+    "status_code": 200,
+    "status_message": "OK",
+    "response": {
+        "Messages": [
+            "C3F8-C3863ED4-334AD4F1"
+        ]
+    }
+}"#;
```

Comment:
this looks so weird, but I guess it has to be formatted like this :)

Reply:
I promise I hate it with all my soul. And if you find better formatting that does not make my OCD jump each time I see it I would be grateful 😅